test/benchmarks/Benchmarks.Trace/CallTarget.cs

```diff
@@ -0,0 +1,733 @@
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Text;
+using Datadog.Trace.ClrProfiler.CallTarget;
+
+namespace Benchmarks.Trace
+{
+    /// <summary>
+    /// Helper class to simulate the calltarget rewriting for benchmark tests
+    /// </summary>
+    internal unsafe static class CallTarget
+    {
+        // ***************************************************************************************************************
+        //  Run
+        // ***************************************************************************************************************
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static TResult Run<TIntegration, TTarget, TResult>(TTarget targetInstance, delegate*<TResult> bodyCallback)
```

Comment:
According to MS docs: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-9.0/function-pointers

> These types are represented using the function pointer type as outlined in ECMA-335. This means invocation of a delegate* will use calli where invocation of a delegate will use callvirt on the Invoke method. Syntactically though invocation is identical for both constructs.

Will the resulting `calli` instruction make our benchmarks appear faster than they normally would be? Could we instead use a normal delegate here? I'm not very familiar with the `calli` IL instruction, but I wonder if there's a way we could do `callvirt` and `call` instructions so it better matches our runtime instrumentation

Reply:
Well the calli here is on purpose because is the fastest one. The reason is because in our instrumentation there is no call at all because we run the actual method body. In this case the method body is outside the calltarget try/catch construct so I need the fastest call I can make. That means that the current calltarget is faster than the actual calli.

Reply:
Oh I see, that makes sense, I forgot that we're already inside the method. Thank you for clarifying!! 😄